packages/web/src/components/common/token-amount-input/TokenAmountInput.tsx

```diff
@@ -44,14 +46,35 @@ const TokenAmountInput: React.FC<TokenAmountInputProps> = ({
   const onChangeAmountInput = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       const value = event.target.value;
+
       if (value === "") {
         changeAmount("");
+        return;
       }
-      if (value !== "" && !isAmount(value)) return;
 
+      if (integersOnly) {
+        if (value.includes(".")) {
+          return;
+        }
```

Comment:
The decimal point is automatically filtered out by the regex `/^\d+$/`, making it an unnecessary check. It seems like it can be removed.